core/src/main/java/com/linecorp/armeria/internal/common/util/ServiceNamingUtil.java

```diff
@@ -32,7 +32,11 @@ public static String fullTypeHttpServiceName(HttpService service) {
                 continue;
             }
             if (delegate instanceof AnnotatedService) {
-                return ((AnnotatedService) delegate).serviceName();
+                final AnnotatedService annotatedService = (AnnotatedService) delegate;
+                if (annotatedService.name() != null) {
+                    return annotatedService.name();
+                }
```

Comment:
Shouldn't call the same method twice unnecessarily, even if it's dirt cheap.
```suggestion
                final String serviceName = annotatedService.name();
                if (serviceName != null) {
                    return serviceName;
                }
```